src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -57,13 +57,11 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
-                _trace.Error(errorText);
-
-                throw new InvalidOperationException(errorText);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid directory location.", _folderSource.Path);
+                var ex = new InvalidOperationException(errorText);
+                _trace.Error(errorText, ex);
```

Comment:
In the context of a standard SDK extension, if the listener for a function cannot start, that shouldn't be ignored - it is an exceptional situation. This really should be an index time error. Is there a way for you to validate this path earlier?